src/index.ts

```diff
@@ -1,15 +1,18 @@
-import { CapabilitiesResponse, Connector, ConnectorError, ExplainResponse, Forbidden, ForeignKeyConstraint, MutationRequest, MutationResponse, NotSupported, ObjectType, QueryRequest, QueryResponse, SchemaResponse, UniquenessConstraint, start } from "@hasura/ndc-sdk-typescript";
+import { BadGateway, CapabilitiesResponse, Connector, ExplainResponse, Forbidden, ForeignKeyConstraint, MutationRequest, MutationResponse, NotSupported, ObjectType, QueryRequest, QueryResponse, SchemaResponse, UniquenessConstraint, start } from "@hasura/ndc-sdk-typescript";
 import mysql, { Pool } from 'mysql2/promise';
 import { readFileSync } from "fs";
 import { CAPABILITIES_RESPONSE } from "./constants";
 import { doGetSchema } from "./handlers/schema";
 import { doQuery } from "./handlers/query";
 import { doQueryExplain } from "./handlers/queryExplain";
 
-const SINGLESTORE_HOST = process.env["SINGLESTORE_HOST"] as string;
-const SINGLESTORE_PORT = process.env["SINGLESTORE_PORT"] as string;
-const SINGLESTORE_USER = process.env["SINGLESTORE_USER"] as string;
-const SINGLESTORE_PASSWORD = process.env["SINGLESTORE_PASSWORD"] as string;
```

Comment:
maybe we can leave these options and use them if `SINGLESTORE_URL` is not set? Sometimes it's more convenient, especially when the password contains special characters.

Reply:
Added them back + added SSL parameters (I copied all parameters from our VSCode extension)